_RFCs/0007-deprecating-packages.md

```diff
@@ -0,0 +1,59 @@
+* [core/jre9](https://github.com/habitat-sh/core-plans/tree/master/jre9)(Oracle has indicated [here](http://www.oracle.com/technetwork/java/javase/downloads/jre9-downloads-3848532.html) that Java SE 9 has reached end of support and users should upgrade to Java SE 10. Additionally, Oracle's download links to JRE 9 and JDK 9 have been removed. It should be noted that Oracle DOES still support JRE 8 and JDK 8.)
+* [core/jdk9](https://github.com/habitat-sh/core-plans/blob/master/jdk9/plan.sh)(see explanation for core/jre9)
+
+When we become aware that a piece of software we package in core plans has been EOL'd or the author/manufacturer otherwise stops supporting it, we will:
+* Open an RFC explaining why the plan must be deprecated
+* When that RFC is approved (though the normal RFC approval process), we must
+1) Delist the package in builder by setting it to "private" (that will make it invisible to anyone outside of the core org, as mentioned [in this issue](https://github.com/habitat-sh/builder/issues/18)
+2) Add a "Deprecated" notice to the core plan's README in the [core plans repo](https://github.com/habitat-sh/core-plans)
+3) Add some sort of indication to the core plan's view page in Builder (remember, even if it is set to private and not viewable publicly, it will still be viewable to members of the core origin) (this will likely require some feature work)
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+See the Guide-level explanation
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This would necessitate users changing their own behavior and the behavior of their software.
+
+It will also necessitate some feature work on the part of Builder - we would need a field of something to that effect indicating when a plan is deprecated.
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+Here are few alternatives:
+
+* Removing unsupported software completely from Builder - this would break anyone who depends on the software and cannot upgrade at this time. If there is at least one successful build of the software, we should preserve it on Builder for those who must use it, while indicating that it is deprecated and users should upgrade as soon as they can. Setting the plan as "private" makes the plan unsearchable, but still downloadable by those who are already using it.
+* Keep a cache of downloaded software packages - this would seem to get us around download links being removed (which is what happened with jre9/jdk9), but would necessitate a lot more storage and feature work on the part of Builder. Even if we did this, it would not get us around the fact that the software is no longer supported by the author/manufacturer.
```

Comment:
Wouldn't this be a requirement in general, not just an alternative? For builder to rebuild a deprecated package when base packages or other dependencies change, it will need to pull the source from somewhere.

Reply:
Hi @wduncanfraser - thank you for the feedback.  Although we have discussed keeping the downloaded source somewhere, we haven't yet been willing to commit to that (partly due to the storage requirements that would entail).  We can definitely discuss that more, but that should be in a separate RFC.

I don't think that deprecated packages should be rebuilt when base dependencies change (especially with ones like ruby22 which simply do not work with the latest version of version of gcc and, since they are no longer supported by the author, will not be updated to work with it.  We can sometimes patch software to work with updated dependencies, but that gets very hacky very fast and introduces a large maintenance burden), rather they should be frozen with the dependencies they had before they were deprecated.

Reply:
That works for me :) I just wanted to clarify. Based on @rsertelon and @adamhjk 's comments above, it seemed like it was desirable that packages would remain usable without pinning after a base refresh.

Reply:
Good to clarify :)